crates/runtime/src/embeddings/connector.rs

```diff
@@ -293,6 +324,54 @@ impl EmbeddingConnector {
     }
 }
 
+fn get_and_validate_partition_by(
+    df_schema: &DFSchema,
+    vector_store: &VectorStore,
+    ctx: &SessionContext,
+) -> Result<Vec<Expr>, ValidationError> {
+    // Expression must use the bucket UDF with a column in the dataset
+    struct BucketCriterion;
```

Comment:
Is there a benefit to defining this struct inline this function vs defining it outside of the function?